netket/driver/vmc_rgn.py

```diff
@@ -0,0 +1,174 @@
+# Copyright 2021 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import jax
+import jax.numpy as jnp
+
+from textwrap import dedent
+
+from netket.utils.types import PyTree, Array
+from netket.operator import AbstractOperator
+from netket.stats import Stats
+from netket.vqs import MCState
+from netket.optimizer import (
+    identity_preconditioner,
+    PreconditionerT,
+)
+from netket.utils import warn_deprecation
+from netket.optimizer import Sgd
+
+from .vmc_common import info
+from .abstract_variational_driver import AbstractVariationalDriver
+
+
+class VMC_RGN(AbstractVariationalDriver):
```

Comment:
If you make this a subclass of VMC then you don't need to redefine everything, you just need to define the new `forward_and_backward`

```suggestion
class VMC_RGN(VMC):
```

Reply:
Yes this will definitely simplify things